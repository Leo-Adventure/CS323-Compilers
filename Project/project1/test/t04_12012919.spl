struct struct_1
{
  int num;
  char value;
  float res[10];
};
int t04_12012919(int num)
{
  struct struct_1 obj;
  obj.num = num;
  obj.res[1] = 2.3;
}
