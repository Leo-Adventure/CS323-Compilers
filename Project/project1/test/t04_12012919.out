Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: struct_1
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: num
              SEMI
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: char
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: value
                SEMI
              DefList (5)
                Def (5)
                  Specifier (5)
                    TYPE: float
                  DecList (5)
                    Dec (5)
                      VarDec (5)
                        VarDec (5)
                          ID: res
                        LB
                        INT: 10
                        RB
                  SEMI
          RC
      SEMI
    ExtDefList (7)
      ExtDef (7)
        Specifier (7)
          TYPE: int
        FunDec (7)
          ID: t04_12012919
          LP
          VarList (7)
            ParamDec (7)
              Specifier (7)
                TYPE: int
              VarDec (7)
                ID: num
          RP
        CompSt (8)
          LC
          DefList (9)
            Def (9)
              Specifier (9)
                StructSpecifier (9)
                  STRUCT
                  ID: struct_1
              DecList (9)
                Dec (9)
                  VarDec (9)
                    ID: obj
              SEMI
          StmtList (10)
            Stmt (10)
              Exp (10)
                Exp (10)
                  Exp (10)
                    ID: obj
                  DOT
                  ID: num
                ASSIGN
                Exp (10)
                  ID: num
              SEMI
            StmtList (11)
              Stmt (11)
                Exp (11)
                  Exp (11)
                    Exp (11)
                      Exp (11)
                        ID: obj
                      DOT
                      ID: res
                    LB
                    Exp (11)
                      INT: 1
                    RB
                  ASSIGN
                  Exp (11)
                    FLOAT: 2.3
                SEMI
          RC
